server/apps/microservices/src/processors/search-index.processor.ts

```diff
@@ -0,0 +1,28 @@
+import { ISearchIndexAlbumJob, ISearchIndexAssetJob, JobName, QueueName, SearchService } from '@app/domain';
+import { Process, Processor } from '@nestjs/bull';
+import { Job } from 'bull';
+
+@Processor(QueueName.SEARCH_INDEX)
+export class SearchIndexProcessor {
+  constructor(private searchService: SearchService) {}
+
+  @Process({ name: JobName.SEARCH_INDEX_ASSETS, concurrency: 1 })
+  async indexAssets() {
+    await this.searchService.indexAssets();
+  }
+
+  @Process({ name: JobName.SEARCH_INDEX_ASSET, concurrency: 1 })
+  async indexAsset(job: Job<ISearchIndexAssetJob>) {
+    await this.searchService.indexAsset(job.data.asset);
+  }
```

Comment:
Can this have higher concurrency, or is there some limit to that on typesense's side? (Or does it race through fast enough that even `concurrency: 1` is done very quickly?)

Reply:
Oh, yeah we can up this. The bulk ones may need to be tweaked before they can be run in parallel.